src/watchdog.rs

```diff
@@ -0,0 +1,18 @@
+
+
+
+/// A watchdog timer to reset the processor if software is frozen or stalled.
+pub trait Watchdog {
```

Comment:
You're going to want to add `#[cfg(feature = "unproven")]` to this, since it's a new trait